superset/datasets/api.py

```diff
@@ -76,10 +76,22 @@ class DatasetRestApi(BaseSupersetModelRestApi):
         "changed_by_url",
         "changed_by.username",
         "changed_on",
+        "columns.id",
+        "column_count",
         "default_endpoint",
         "explore_url",
+        "kind",
+        "metrics.id",
```

Comment:
Required to join the `metrics` to get a count. I'm looking into a FAB change to make this less hacky.

Reply:
cc @dpgaspar  does that result in round trips for each entry in the page? It seems like a bad idea. Is number of cols/metrics necessary here? Do we have a wireframe for context?

Reply:
This should not result in an N+1, given some recent work that Daniel has done. I'll update the ticket with a screenshot of the design.

Reply:
In general, adding a count of related records is a common problem for webapps and there are performant solutions. I can see us wanting to display similar information in other places, so I recommend we find a performant way forward with FAB.

Reply:
@willbarrett is preparing a PR on FAB to add the ability to define a custom SQLAlchemy query for the list endpoint, should be a great new feature